src/katgpucbf/fsim/main.py

```diff
@@ -0,0 +1,330 @@
+################################################################################
+# Copyright (c) 2024, National Research Foundation (SARAO)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""
+Simulate channelised data from the MeerKAT F-Engines destined for one or more XB-Engines.
+
+Refer to :ref:`feng-packet-sim` for more information.
+"""
+
+import argparse
+import asyncio
+import functools
+import itertools
+import os
+import time
+from typing import Sequence
+
+import katsdpservices
+import numpy as np
+import prometheus_async
+import spead2.send.asyncio
+from katsdptelstate.endpoint import endpoint_list_parser
+from numba import njit
+from prometheus_client import Counter, Gauge
+
+from .. import (
+    BYTE_BITS,
+    COMPLEX,
+    DEFAULT_JONES_PER_BATCH,
+    DEFAULT_PACKET_PAYLOAD_BYTES,
+    DEFAULT_TTL,
+    N_POLS,
+    SPEAD_DESCRIPTOR_INTERVAL_S,
+    spead,
+)
+from ..fgpu.send import PREAMBLE_SIZE, make_descriptor_heap
+from ..send import DescriptorSender
+from ..utils import TimeConverter, add_gc_stats, comma_split
+from . import METRIC_NAMESPACE
+
+DTYPE = np.dtype(np.int8)
+#: Number of heaps in time to keep in flight
+QUEUE_DEPTH = 8
+output_heaps_counter = Counter("output_heaps", "number of heaps transmitted", namespace=METRIC_NAMESPACE)
+output_bytes_counter = Counter("output_bytes", "number of payload bytes transmitted", namespace=METRIC_NAMESPACE)
+time_error_gauge = Gauge(
+    "time_error_s", "elapsed time minus expected elapsed time", ["stream"], namespace=METRIC_NAMESPACE
+)
+
+
+def parse_args(arglist: Sequence[str] | None = None) -> argparse.Namespace:
+    """Parse the command-line arguments."""
+    parser = argparse.ArgumentParser(prog="fsim")
+
+    parser.add_argument(
+        "--adc-sample-rate", type=float, default=1712e6, help="Digitiser sampling rate (Hz) [%(default)s]"
+    )
+    parser.add_argument("--interface", default="lo", help="Network interface on which to send packets [%(default)s]")
+    parser.add_argument("--array-size", type=int, default=80, help="Number of antennas in the array")
+    parser.add_argument(
+        "--channels", type=int, default=32768, help="Total number of channels in the array [%(default)s]"
```

Comment:
bike-shedding:

```suggestion
        "--channels", type=int, default=32768, help="Total number of channels in the (simulated) array [%(default)s]"
```

Reply:
Done (but without parentheses) and also applied to `--array-size` for consistency.